### DISCUSSION TITLE
RelationManager attaching to a model that has a related table

### DISCUSSION THREAD

**Discussion by seb-ogsi** [filamentphp/filament#8082] [Help]
Hi

I'd like to customize the attach select that is shown when i attach a resource to a course.

The model to attach to has a relation to a related table (like a category of the resource). 

Table **COURSE**
...

Table **COURSE_RESOURCE**
course_id
resource_id

Table **RESOURCE**
resource_name
resource_type_id
....

Table **RESOURCE_TYPE**
resource_type

When attaching a resource to a course with the relationManager, i'd like to be able to use the select to filter on both the resource_name AND the resource_type. Or at least showing the resource_type with the resource_name and not only the resource_name.

The issue is that i have multiple same resource_name that have different resource_type. Right now, i could search by resource_name and their are all showed but i can tell wich select option is related to which resource_type.

I try this (where resourceType is the eloquent model relation belongsTo between the 2 tables resource && resource_type)

AttachAction::make()
    ->recordSelectSearchColumns(['resource_name', 'resourceType.resource_type'])

but i doesn't work, it does not make the join in the query. 

An other way would be to present a full table of the model to attach to instead of just the select.

Is there a way to setup the attach form to show a full table of resource (with related resource_type column) with filter so that you can select any row to attach to (after eventualy filtering) ? 

Any idea ?

**Comment by Rattone** [comment]
Same problem here with recordSelectSearchColumns and relational fields.
I think it's a common behavior, in my case I want to search for a user by email (user table) and by first_name/last_name (profile table)

**Reply by seb-ogsi** [reply]
I haven't found any solution yet so i'm going for a **custom filament page** with **custom livewire component** and i'll try to manage the attach out of filament **relationManager** when it's not a classical relation (those work very well with **relationManager**)

**Comment by ashokssrkc** [comment]
Hey guys, Any luck solving this? My self-stuck here in showing the same relation field value!
